lit_tests/kernel/wave/gemm.py

```diff
@@ -2322,3 +2322,139 @@ def repeat(acc: tkl.Register[M, N, tkl.f32]) -> tkl.Register[M, N, tkl.f32]:
     # CHECK:              amdgpu.mfma
     # Verify write to global memory (outside loop)
     # CHECK:            vector.store %{{.*}}, %[[GLOBAL_C]]
+
+
+@run_test
+def test_persistent_gemm():
+    TOTAL_TILES = tkl.sym.TOTAL_TILES
+    NUM_CTAS = tkl.sym.NUM_CTAS
+    TILE_IDX = tkl.sym.TILE_IDX
+    CTA_M_OFFSET = tkl.sym.CTA_M_OFFSET
+    CTA_N_OFFSET = tkl.sym.CTA_N_OFFSET
+    N_TILES = tkl.sym.N_TILES
+
+    i = tkw.IndexMapping.iterator(0)
+    j = tkw.IndexMapping.iterator(1)
+
+    a_read_mapping = tkw.IndexMapping(
+        num_iterators=2,
+        inputs={M: i + CTA_M_OFFSET, K: j},
+        outputs={M: i, K: j},
+    )
+
+    b_read_mapping = tkw.IndexMapping(
+        num_iterators=2,
+        inputs={N: i + CTA_N_OFFSET, K: j},
+        outputs={N: i, K: j},
+    )
+
+    c_write_mapping = tkw.IndexMapping(
+        num_iterators=2,
+        inputs={M: i, N: j},
+        outputs={M: i + CTA_M_OFFSET, N: j + CTA_N_OFFSET},
+    )
+
+    constraints: list[tkw.Constraint] = [
+        tkw.GridConstraint(NUM_CTAS),
+        tkw.WorkgroupConstraint(M, BLOCK_M, 0, primary=False),
+        tkw.WorkgroupConstraint(N, BLOCK_N, 0),
+        tkw.TilingConstraint(K, BLOCK_K),
+        tkw.TilingConstraint(TILE_IDX),
+        tkw.WaveConstraint(M, BLOCK_M / 2),
+        tkw.WaveConstraint(N, BLOCK_N / 2),
+        tkw.HardwareConstraint(
+            threads_per_wave=64,
+            mma_type=tkw.MMAType.F32_16x16x16_F16,
+            vector_shapes={TILE_IDX: 0},
+            use_linearized_cta_dims=True,
+        ),
+    ]
+
+    @tkw.wave(constraints)
+    def persistent_gemm(
+        a: tkl.Memory[M, K, ADDRESS_SPACE, tkl.f16],
+        b: tkl.Memory[N, K, ADDRESS_SPACE, tkl.f16],
+        c: tkl.Memory[M, N, GLOBAL_ADDRESS_SPACE, tkl.f32],
+    ):
+        total_tiles_scalar = tkw.scalar(TOTAL_TILES, tkl.i32)
+        tkw.set_symbol(TOTAL_TILES, total_tiles_scalar)
```

Comment:
This part is a little confusing. Why first create a scalar based on TOTAL_TILES and then set TOTAL_TILES to total_tiles_scalar? Why is this needed?